### PR TITLE
Move AnalyzerOptionsProvider to Analyzers layer

### PR DESCRIPTION
**Pull request by Youssef1313**
The corresponding language-specific types are already in Analyzers layer.

https://github.com/dotnet/roslyn/blob/main/src/Analyzers/CSharp/Analyzers/CodeStyle/CSharpAnalyzerOptionsProvider.cs

So I thought `AnalyzerOptionsProvider` should be in Analyzers too.

### CODE REVIEW

**Comment by Youssef1313** [comment]
@mavasani @tmat Can you take a look please? Thanks!

**Comment by Youssef1313** [comment]
Ping @mavasani @tmat for review.

File: src/Analyzers/Core/CodeFixes/CodeFixes.projitems [hunk 1]
```diff
@@ -22,6 +22,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)AddObsoleteAttribute\AbstractAddObsoleteAttributeCodeFixProvider.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)AddRequiredParentheses\AddRequiredParenthesesCodeFixProvider.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)AliasAmbiguousType\AbstractAliasAmbiguousTypeCodeFixProvider.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)AnalyzerOptionsProviders.cs" />
```
Comment (mavasani) [inline src/Analyzers/Core/CodeFixes/CodeFixes.projitems:25]: Is there a reason why we had this items both in the CompilerExtensions and WorkspaceExtensions layers? I think having it in the former makes the inclusion in the latter redundant. Am I missing something?

Comment (Youssef1313) [reply src/Analyzers/Core/CodeFixes/CodeFixes.projitems:25]: We didn't have it in both. There are two different files, `AnalyzerOptionsProviders` and `AnalyzerOptionsProvider`

Comment (mavasani) [reply src/Analyzers/Core/CodeFixes/CodeFixes.projitems:25]: Ah, I missed that difference!!

**Review by mavasani — APPROVED** [review]

**Pull request merged**